packages/app/src/heartbeatService.ts

```diff
@@ -303,3 +307,27 @@ export function countBytes(heartbeatsCache: HeartbeatsByUserAgent[]): number {
     JSON.stringify({ version: 2, heartbeats: heartbeatsCache })
   ).length;
 }
+
+/**
+ * Returns the index of the heartbeat with the earliest date.
+ * If the heartbeats array is empty, -1 is returned.
+ */
+export function getEarliestHeartbeatIdx(
```

Comment:
Can the array be out of chronological order?

Reply:
I think it could be out of order.

To guarantee that the objects retrieved from IndexedDB are returned in insertion order (chronological order) we need to use an index https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/index